scripts/linters/pylint_extensions_test.py

```diff
@@ -2234,3 +2234,191 @@ def test_file_overview_at_end_of_file(self):
+        filename = temp_file.name
+
+        with python_utils.open_file(filename, 'w') as tmp:
+            tmp.write(
+                u"""
+                    class ClassName(Class):
+                        def func():
+                            \"\"\"This is a docstring.\"\"\"
+                            a = b + c
+                """)
+        node_class_with_no_docstring.file = filename
+        node_class_with_no_docstring.path = filename
+
+        self.checker_test_object.checker.process_module(
+            node_class_with_no_docstring)
+
+        with self.checker_test_object.assertNoMessages():
+            temp_file.close()
+
+    def test_class_with_no_body(self):
+        node_class_with_no_body = (
+            astroid.scoped_nodes.Module(
+                name='test',
+                doc='Custom test'))
+        temp_file = tempfile.NamedTemporaryFile()
+        filename = temp_file.name
+
+        with python_utils.open_file(filename, 'w') as tmp:
+            tmp.write(
+                u"""
+                    class ClassName(Class):
+                        \"\"\"This is a docstring.\"\"\"
+
+                    def func():
+                        a = 1 + 2
+                """)
+        node_class_with_no_body.file = filename
+        node_class_with_no_body.path = filename
+
+        self.checker_test_object.checker.process_module(
+            node_class_with_no_body)
+
+        with self.checker_test_object.assertNoMessages():
+            temp_file.close()
+
+    def test_none_and_excessive_newline_below_class_docstring(self):
+        node_none_and_excessive_newline_below_class_docstring = (
+            astroid.scoped_nodes.Module(
+                name='test',
+                doc='Custom test'))
+        temp_file = tempfile.NamedTemporaryFile()
+        filename = temp_file.name
+
+        with python_utils.open_file(filename, 'w') as tmp:
+            tmp.write(
+                u"""
+                    class ClassName(Class):
+                        \"\"\"This is a docstring.\"\"\"
+
+                        a = 1 + 2
+
+                    class ClassName(Class):
+                        \"\"\"This is a docstring.\"\"\"
+                        a = 1 + 2
+
+                    class ClassName(Class):
+                        \"\"\"This is a docstring.\"\"\"
+
+
+                        a = 1 + 2
+                """)
+        node_none_and_excessive_newline_below_class_docstring.file = filename
+        node_none_and_excessive_newline_below_class_docstring.path = filename
+
+        self.checker_test_object.checker.process_module(
+            node_none_and_excessive_newline_below_class_docstring)
+
+        message_with_no_newline = testutils.Message(
+            msg_id='newline-below-class-docstring',
+            line=9)
+
+        message_with_excessive_newline = testutils.Message(
+            msg_id='newline-below-class-docstring',
+            line=15)
```

Comment:
The error message will be `'No single newline below class docstring.`? If yes the I don't think this is a correct message*

Reply:
Changed to `Missing single newline below class docstring.`

Reply:
This is still not correct should we through a different error for this case? Please check `BlankLineBelowFileOverviewChecker` for reference!

Reply:
That check was just to check if there are more than 2 lines between two function definitions, but here we are also checking if there is not even a single newline between class docstring and rest of the class body